src/Plugin/Block/Product/ListProduct.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tweakwise\AttributeLandingTweakwise\Plugin\Block\Product;
+
+use Emico\AttributeLanding\Api\LandingPageRepositoryInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Registry;
+use Tweakwise\Magento2Tweakwise\Block\Product\ListProduct as Subject;
+use Tweakwise\Magento2Tweakwise\Helper\Cache;
+
+class ListProduct
+{
+    /**
+     * @param RequestInterface $request
+     * @param Cache $cacheHelper
+     * @param LandingPageRepositoryInterface $landingPageRepository
+     * @param Registry $registry
+     */
+    public function __construct(
+        private readonly RequestInterface $request,
+        private readonly Cache $cacheHelper,
+        private readonly LandingPageRepositoryInterface $landingPageRepository,
+        private readonly Registry $registry
+    ) {
+    }
+
+    /**
+     * @param Subject $subject
+     * @param string $route
+     * @param array $params
+     * @return array
+     */
+    public function beforeGetUrl(Subject $subject, $route = '', $params = [])
+    {
+        $landingPageId = (int)$this->request->getParam('id');
+        if (
+            !$this->isAttributeLandingRequest() ||
+            !$landingPageId ||
+            !$this->cacheHelper->personalMerchandisingCanBeApplied() ||
+            $route !== 'page_cache/block/esi'
+        ) {
+            return [$route, $params];
+        }
+
+        try {
+            $landingPage = $this->landingPageRepository->getById($landingPageId);
+        } catch (NoSuchEntityException | LocalizedException $e) {
+            return [$route, $params];
+        }
+
+        $filters = $this->registry->registry(sprintf('alp_filters_%s', $landingPageId)) ?? [];
```

Comment:
@jansentjeu I think this can be changed to $filters = $landingPage->getFilters(). So the registery change is not needed anymore.